src/pages/Homepage.js

```diff
@@ -0,0 +1,94 @@
+import allStockImg from '../components/images/all_stock.svg';
+import consumerCyclical from '../components/images/consumer-svgrepo-com.svg';
+import energy from '../components/images/energy-svgrepo-com.svg';
+import technology from '../components/images/Technology.svg';
+import industrial from '../components/images/industry-svgrepo-com.svg';
+import financialServices from '../components/images/financial-services.svg';
+import basicMaterials from '../components/images/basic-home-svgrepo-com.svg';
+import communicationServices from '../components/images/communication-svgrepo-com.svg';
+import consumerDefence from '../components/images/consumer-defence.svg';
+import healthCare from '../components/images/healthcare-svgrepo-com.svg';
+import realEstate from '../components/images/real-estate-svgrepo-com.svg';
+import utilities from '../components/images/Utilities.svg';
+import industrialGoods from '../components/images/Industrial_Goods.svg';
+import financial from '../components/images/Financial.svg';
+import services from '../components/images/Services.svg';
+import conglomerates from '../components/images/Conglomerates.svg';
+
+const Homepage = () => {
+  const allStocks = useSelector((state) => state.allStocks.data);
+  const loading = useSelector((state) => state.allStocks.loading);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (allStocks.allAssets.length < 1) {
+      dispatch(fetchStocks());
+    }
+  }, [dispatch, allStocks]);
+
+  const filterOptions = [
+    { query: 'Consumer Cyclical', img: consumerCyclical },
+    { query: 'Energy', img: energy },
+    { query: 'Technology', img: technology },
+    { query: 'Industrials', img: industrial },
+    { query: 'Financial Services', img: financialServices },
+    { query: 'Basic Materials', img: basicMaterials },
+    { query: 'Communication Services', img: communicationServices },
+    { query: 'Consumer Defensive', img: consumerDefence },
+    { query: 'Healthcare', img: healthCare },
+    { query: 'Real Estate', img: realEstate },
+    { query: 'Utilities', img: utilities },
+    { query: 'Industrial Goods', img: industrialGoods },
+    { query: 'Financial', img: financial },
+    { query: 'Services', img: services },
+    { query: 'Conglomerates', img: conglomerates },
+  ];
+
+  return (
+    <div className="homepage">
+      {loading ? <Loader /> : null}
+      <div className="all__stock__container">
+        <img src={allStockImg} alt="" />
+        <div>
+          <h2>All Stock Data On The Stock Market</h2>
+          <span>
+            {allStocks.allAssets.length.toLocaleString()}
+            {' '}
+            Total Stocks
+          </span>
+          <Link to="/listall">
+            See All Stocks
+            {' '}
+            <i className="fa-sharp fa-solid fa-arrow-right" />
+            {' '}
+          </Link>
+        </div>
+      </div>
+      <h4>Filter By Sector:</h4>
+      <div className="filter__container">
+        {filterOptions.map((val) => (
```

Comment:
- [ ] Based on the requirements of this project, which say that the Homepage should have a list of categories(Let's kindly display a list of stocks on the homepage :+1: ). These stocks should be filtered (We shall learn to JavaScript ES6 filter() method) using a given parameter(You can use the stock name). We shall use the **Filter** stateless component to filter data retrieved from the API.


NB: You can use a search field on the homepage. Inside the search field, a user can introduce the beginning of the name of a stock and the filter will get all stocks with those letters.